diff --git a/configure b/configure
index f76f946dfe..fa11da5147 100755
--- a/configure
+++ b/configure
@@ -352,6 +352,7 @@ External library support:
   --enable-libmfx          enable Intel MediaSDK (AKA Quick Sync Video) code via libmfx [no]
   --enable-libvpl          enable Intel oneVPL code via libvpl if libmfx is not used [no]
   --enable-libnpp          enable Nvidia Performance Primitives-based code [no]
+  --disable-libnpp-static  disable using static version of libnpp [autodetect]
   --enable-mmal            enable Broadcom Multi-Media Abstraction Layer (Raspberry Pi) via MMAL [no]
   --disable-nvdec          disable Nvidia video decoding acceleration (via hwaccel) [autodetect]
   --disable-nvenc          disable Nvidia video encoding code [autodetect]
@@ -2021,6 +2022,7 @@ HWACCEL_LIBRARY_NONFREE_LIST="
     cuda_nvcc
     cuda_sdk
     libnpp
+    libnpp_static
 "
 
 HWACCEL_LIBRARY_LIST="
@@ -3344,6 +3346,13 @@ overlay_cuda_filter_deps="ffnvcodec"
 overlay_cuda_filter_deps_any="cuda_nvcc cuda_llvm"
 sharpen_npp_filter_deps="ffnvcodec libnpp"
 
+if ! disabled libnpp_static; then
+    scale_npp_filter_deps="ffnvcodec libnppif_static"
+    scale2ref_npp_filter_deps="ffnvcodec libnppif_static"
+    transpose_npp_filter_deps="ffnvcodec libnppif_static"
+    sharpen_npp_filter_deps="ffnvcodec libnppif_static"
+fi
+
 ddagrab_filter_deps="d3d11va IDXGIOutput1 DXGI_OUTDUPL_FRAME_INFO"
 
 amf_deps_any="libdl LoadLibrary"
@@ -6994,8 +7003,14 @@ enabled libmodplug        && require_pkg_config libmodplug libmodplug libmodplug
 enabled libmp3lame        && require "libmp3lame >= 3.98.3" lame/lame.h lame_set_VBR_quality -lmp3lame $libm_extralibs
 enabled libmysofa         && { check_pkg_config libmysofa libmysofa mysofa.h mysofa_neighborhood_init_withstepdefine ||
                                require libmysofa mysofa.h mysofa_neighborhood_init_withstepdefine -lmysofa $zlib_extralibs; }
-enabled libnpp            && { check_lib libnpp npp.h nppGetLibVersion -lnppig -lnppicc -lnppc -lnppidei -lnppif ||
-                               check_lib libnpp npp.h nppGetLibVersion -lnppi -lnppif -lnppc -lnppidei ||
+enabled libnpp            && { { if ! disabled libnpp_static; then
+                                 check_lib libnpp npp.h nppGetLibVersion -lnppig_static -lnppim_static -lnppist_static -lnppicc_static -lnppial_static -lnppc_static -lnppidei_static -lnppif_static -lcudart_static -lculibos $ldl -lstdc++ -lrt ||
+                                 check_lib libnpp npp.h nppGetLibVersion -lnppif_static -lnppim_static -lnppist_static -lnppif_static -lnppc_static -lnppidei_static -lcudart_static -lculibos $ldl -lstdc++ -lrt
+                               else false; fi; } ||
+                               { if ! enabled libnpp_static; then
+                                 check_lib libnpp npp.h nppGetLibVersion -lnppig -lnppicc -lnppc -lnppidei -lnppif ||
+                                 check_lib libnpp npp.h nppGetLibVersion -lnppi  -lnppif  -lnppc -lnppidei
+                               else false; fi; } ||
                                die "ERROR: libnpp not found"; }
 enabled libopencore_amrnb && require libopencore_amrnb opencore-amrnb/interf_dec.h Decoder_Interface_init -lopencore-amrnb
 enabled libopencore_amrwb && require libopencore_amrwb opencore-amrwb/dec_if.h D_IF_init -lopencore-amrwb
